Add blog post for improvements to data access

GeigerJ2 opened the pull request:
Meaning the process dumping feature and the more intuitive QB syntax.

EDIT: This should be ready for a first round of review. Thanks to @agoscinski's wisdom, I just changed the base of this PR to the branch of PR #69 on which it depends, so only the actual changes are shown. Requested review from @edan-bainglass as the author of the new QB syntax.

[Direct link](https://aiida-website--72.org.readthedocs.build/news/posts/2024-10-04-data-access.html) to the rendered version on RTD.

agoscinski commented on create_post.py:
```diff
@@ -13,6 +13,7 @@
     "event": "Events",
     "report": "Reports",
     "release": "Releases",
+    "blog": "Blog",
```
not related, but I added it, here since it is the first blog post

agoscinski commented on docs/news/posts/2024-10-04-data-access.md:
```diff
@@ -0,0 +1,283 @@
+blogpost: true
+category: Blog
+tags: usability
+date: 2024-10-04
```
This is not shown because the date is in the future. You have to import from `__future__`

agoscinski commented on docs/news/posts/2024-10-04-data-access.md:
````diff
@@ -0,0 +1,283 @@
+├── 03-scf-PwBaseWorkChain
+│  ├── ...
+...
+```
````
For me these render weird (see image it the end)

GeigerJ2 replied:
Should be fixed now!

agoscinski commented on docs/news/posts/2024-10-04-data-access.md:
````diff
@@ -0,0 +1,283 @@
+to the more concise:
+
+```python
+filters=(Int.fields.pk > 1) & (Int.fields.pk < 10),
````
maybe
```suggestion
filters=(Int.fields.pk > 1) & (Int.fields.pk < 10)
```

GeigerJ2 replied:
That's what happens when I don't ruff-autoformat the shit out of any code snippet I write :D

agoscinski commented on docs/news/posts/2024-10-04-data-access.md:
```diff
@@ -0,0 +1,283 @@
+        {"pk": {">": 1}},
+        {"pk": {"<": 10}},
+    ]
+},
```
maybe
```suggestion
}
```

agoscinski commented on docs/news/posts/2024-10-04-data-access.md:
```diff
@@ -0,0 +1,283 @@
+[^1]:
+  The file repository is based on the [`disk-objectstore`](https://github.com/aiidateam/disk-objectstore)
+  implementation. If you ever wondered what the `_dos` appendix of the `core.psql_dos` and `core.sqlite_dos` storage
+  backends means, now you know! 😉
```
obviously it is disk operating system

agoscinski commented on docs/news/posts/2024-10-04-data-access.md:
```diff
@@ -0,0 +1,283 @@
+  The file repository is based on the [`disk-objectstore`](https://github.com/aiidateam/disk-objectstore)
+  implementation. If you ever wondered what the `_dos` appendix of the `core.psql_dos` and `core.sqlite_dos` storage
+  backends means, now you know! 😉
+
```
Footnotes do not work properly. Maybe the linebreak?

GeigerJ2 replied:
Hm, weird, they worked fine in the other blog post despite the linebreaks. Will have a look!

GeigerJ2 replied:
Was the indent that was wrong...

agoscinski reviewed (commented):
Maybe you can zoom into the terminal and use a high contrast mode? It is quite hard to see what is happening in the terminal.

This is not rendered properly
<img width="1194" alt="Screenshot 2024-09-17 at 15 42 04" src="https://github.com/user-attachments/assets/9fa7e05f-0a70-4c88-b412-1dd50cc76f23">

agoscinski reviewed (changes requested):
request changes

GeigerJ2 commented:
> Maybe you can zoom into the terminal and use a high contrast mode? It is quite hard to see what is happening in the terminal.

Yeah, just like in presentations, terminal with white background might actually be preferable here (despite the eyeball cancer :D)

> This is not rendered properly 

Hm, weird, locally it aligns nicely. I'll investigate!

agoscinski reviewed (approved):
Are the `calculation-dump*gif` used somewhere? Can you remove them if not used before merging

edan-bainglass reviewed (dismissed):
LGTM, at least w.r.t new `QueryBuilder` syntax section. Thanks :)

GeigerJ2 commented:
OK, I just squashed the previous commits to clean up here a bit, so this is ready for a final review, @agoscinski, @edan-bainglass, @giovannipizzi.

[Direct link to rendered version on RTD](https://aiida-website--72.org.readthedocs.build/news/posts/2024-10-04-data-access.html)

giovannipizzi commented on docs/news/posts/2024-10-04-data-access.md:
```diff
@@ -0,0 +1,226 @@
+    backends means, now you know! 😉
+
+[^2]:
+    While your workflows are being executed on the remote computer, files are located in a three level hierarchical
```
I would make sure this does not become confusing - the paragraphs above all refer to internal AiiDA storage, here to the scratch.
E.g. start with something like "The discussion above refers to the files and data stored by AiiDA on the computer where AiiDA is installed, and that is preserved in the long term. These files are e.g. retrieved from the remote computer once a calculation finishes, or could e.g. be parsed data or input data provided by the user. Instead, while your workflows ..." (maybe replace workflows with calculations? only those are actually on the remote).

Maybe the sentence could be changed a bit along these lines:

Instead, the execution of calculations on the remote computer are preformed in a subfolder of the `work_directory` of the given `Computer` (typically in the `scratch`), where the subfolder name is generated from the UUID of the AiiDA `CalculationNode`. This directory has a three-level depth, obtained by "sharding" the UUID as follows: if the UUID is for instance `6861d8fb-4694-46be-b0e6-7282989f069d`, the calculation will run in a subfolder named `68/61/d8fb-4694-46be-b0e6-7282989f069d`.

giovannipizzi commented on docs/news/posts/2024-10-04-data-access.md:
```diff
@@ -0,0 +1,226 @@
+    automatically determined and based on the iteration counter, the link label, and the class name.
+
+[^5]:
+    The `dump`ing feature is currently still actively developed to enable obtaining remote and stashed data entities (e.g., intermediate files of the workflow that weren't originally retrieved from the HPC), as well as allowing for
```
As a general comment, I don't know if using backticks as part of a longer work makes things readable. I would just write

The dumping feature

or

The `verdi process dump` feature

or e.g.

The dumping feature (via `verdi process dump`)

but avoid e.g. `dump`ing or similar - if you agree, I would change also the rest

giovannipizzi commented on docs/news/posts/2024-10-04-data-access.md:
```diff
@@ -0,0 +1,226 @@
+[database](https://aiida.readthedocs.io/projects/aiida-core/en/v2.6.2/topics/storage.html), as well as an internal [file
+repository](https://aiida.readthedocs.io/projects/aiida-core/en/stable/topics/repository.html#repository) [^1] to store
+your data locally [^2].
+Both are optimized towards high performance and the difference between the typical file-system approach most of us are
+familiar with, and AiiDA's machine-readable internal data storage can make it difficult to get your data out of it in an
+easily accessible way.
```
```suggestion
Both are optimized towards high performance. The difference between the typical file-system approach (that most of us are
familiar with) and AiiDA's machine-readable internal data storage could make it cumbersome to get your data.
```

giovannipizzi commented on docs/news/posts/2024-10-04-data-access.md:
```diff
@@ -0,0 +1,226 @@
+familiar with, and AiiDA's machine-readable internal data storage can make it difficult to get your data out of it in an
+easily accessible way.
+
+Therefore, you, the user, is effectively forced to use the `verdi` CLI interface or AiiDA's Python API (e.g. the `QueryBuilder`) to access your data, making
```
```suggestion
Therefore, you, the user, are effectively forced to use the `verdi` CLI interface or AiiDA's Python API (e.g. the `QueryBuilder`) to access your data, making
```

giovannipizzi commented on docs/news/posts/2024-10-04-data-access.md:
```diff
@@ -0,0 +1,226 @@
+
+![PwCalculation dump](./_gifs/calculation-dump-white-10fps-2160p.gif)
+
+And for a more complex `PwBandsWorkChain` [^3]:
```
I would put the content of the footnote directly here and not as a footnote?

giovannipizzi commented on docs/news/posts/2024-10-04-data-access.md:
```diff
@@ -0,0 +1,226 @@
+
+![PwBandsWorkChain dump](./_gifs/workflow-dump-white-10fps-2160p.gif)
+
+As you can see, the command works for both, individual calculations, as well as nested workflows [^4], resulting in
+the following output directories:
```
```suggestion
As you can see, the command works both for individual calculations and for nested workflows [^4], resulting in
the following output directories:
```

giovannipizzi commented on docs/news/posts/2024-10-04-data-access.md:
````diff
@@ -0,0 +1,226 @@
+```
+
+Therefore, after running the command once, you'll have all data involved in the execution of your workflow directly
+accessible, and can explore it with the Unix command line utilities we all know and love [^5].
````
```suggestion
accessible, and can explore it as a standard folder [^5].
```
I'm changing this not only to make the sentence easier to read, but also because in principle you are not forced anymore to use a CLI, you can just open it in a file-browser window.

edan-bainglass replied:
Also, "Unix command line utilities we all know and love" a bit biased 😅

GeigerJ2 replied:
@edan-bainglass ofc, get out with your Windows machine :D ❤️

giovannipizzi commented on docs/news/posts/2024-10-04-data-access.md:
```diff
@@ -0,0 +1,226 @@
+Therefore, after running the command once, you'll have all data involved in the execution of your workflow directly
+accessible, and can explore it with the Unix command line utilities we all know and love [^5].
+
+So happy grepping!
```
Depending on whether you apply my change above, you need to adapt this or slightly clarify it. For instance "So, e.g., you can simply use `grep` to search for strings in the input and output files of the calculations!

giovannipizzi commented on docs/news/posts/2024-10-04-data-access.md:
````diff
@@ -0,0 +1,226 @@
+    filters={"and": [{"pk": {">": 1}}, {"pk": {"<": 10}}]},
+    project=["pk", "attributes.value"],
+)
+```
````
Since not all might immediately understand what is going on, maybe you can briefly explain it? E.g. "after importing the needed modules in lines 1-2, we create a QueryBuilder object in line 4. In line 5-12 we filter for a group..."

giovannipizzi commented on docs/news/posts/2024-10-04-data-access.md:
```diff
@@ -0,0 +1,226 @@
+
+as it allows for autocompletion.
+
+Any feedback on the new QB syntax welcome!
```
We never use `QB` above, spell it out

giovannipizzi reviewed (commented):
Some comments, and suggestions to slightly simplify some sentences

edan-bainglass reviewed (approved):
Once Giovanni's comments are addressed, this is good to go from my end.

GeigerJ2 reviewed (commented):
OK, thanks for the review, @giovannipizzi. I have implemented all of your suggested changes, and further improved the text.

GeigerJ2 commented:
@khsrali linkcheck failing on some scientific URLs?

khsrali commented:
I checked the failing url manually:
https://journals.aps.org/prb/abstract/10.1103/PhysRevB.102.024106

And it works.
Most be some sort of blocking problem from GitHub client 🤔

agoscinski commented:
Maybe this works
https://github.com/sphinx-doc/sphinx/issues/10343#issuecomment-1097430133

Would put it in ignore if this does not work, one can use also the doi instead, that should never fail anyway

GeigerJ2 commented:
@khsrali I'm just a bit confused because I guess it was working before. @agoscinski, good idea, I replaced it with the DOIs.

I also removed the `go-fair.org` link, because that website seems to be currently down. It was used in a 4-year-old post about Materials Cloud Archive, so I suppose that should be fine. Though, now https://www.max-centre.eu and https://www.intersect-project.eu also don't work (neither in my browser), so maybe there is currently another CloudFlare-like situation and websites are down 😅
